app/screens/more_dms/more_dms.js

```diff
@@ -446,11 +449,23 @@ export default class MoreDirectMessages extends PureComponent {
                 }
 
                 return true;
-            });
+            };
+
+            let results;
+            if (isGuest) {
+                results = filterProfilesMatchingTerm(searchResults.filter(this.filterUnknownUsers), term).filter(filterByTerm);
```

Comment:
There was already 2 iterations, because the `filterProfileMatchingTerm` is place in another file, we could convert that into a builder of filter function, and use the `.filter` call here instead, but I think, because this is a bug for 1.31 we should keep this with the less number of changes possible.

Reply:
Anyway, I changed it to keep this in 2 iterations instead of 3 for the guest accounts.